Naive greedy alg/scheduling/Scheduler.py

```diff
@@ -123,33 +114,47 @@ def update_schedule(self, time):
             self.req_queue.pop()
             av_servers = [server for server in av_servers if server not in job_servers]
 
-        # Add config value to only consider jobs with data < conf_value
-        jobs_by_mass = sorted(
-            self.active_jobs, key=methodcaller("remaining_mass", time)
-        )
-        while jobs_by_mass and av_servers:
-            job = jobs_by_mass[0]
-            if self._is_job_reconfigurable(job, av_servers, time):
-                av_servers = self._reconfigure_job(job, av_servers, time)
-            jobs_by_mass.pop(0)
-
-        for server in list(av_servers):
-            if not self._shutdown_server(av_servers):
-                break
-            if (
-                0.5
-                < ((len(av_servers) / len(self.servers)) ** self.conf.shutdown_weight)
-                * self.conf.shutdown_prob
-            ):
-                if random() < self.conf.shutdown_time_prob:
-                    shutdown_duration = self.conf.shutdown_time_1
+        # Reconfiguration
+        if self.reconfig_enable:
+            jobs_by_mass = sorted(
+                self.active_jobs, key=methodcaller("remaining_mass", time)
+            )
+            while jobs_by_mass and av_servers:
+                job = jobs_by_mass[0]
+                if self._is_job_reconfigurable(job, av_servers, time):
+                    av_servers = self._reconfigure_job(job, av_servers, time)
+                jobs_by_mass.pop(0)
+
+        # Turn off servers
+        if self.power_off_enable:
+            for server in list(av_servers):
+                if self.param_enable:
+                    if not self._shutdown_server(av_servers):
+                        break
+                    if (
+                        0.5
+                        < (
+                            (len(av_servers) / len(self.servers))
+                            ** self.conf.shutdown_weight
+                        )
+                        * self.conf.shutdown_prob
+                    ):
+                        if random() < self.conf.shutdown_time_prob:
+                            shutdown_duration = self.conf.shutdown_time_1
+                        else:
+                            shutdown_duration = self.conf.shutdown_time_2
```

Comment:
We could move this in a separate function and merge the shutdown task creation code
The function could return a boolean (indicating if the server can be shut down) and the appropriate shutdown_duration
This will allow having a clearer workflow.

Smth along these lines:

        # Turn off servers
        if self.power_off_enable:
            for server in list(av_servers):
                if not self._shutdown_server(av_servers):
                    break

                shutdown, duration = self._allow_shutdown(av_servers)
                if shutdown:
                    power_off = Job.make_power_off(
                        [server], start_time=time, duration=duration
                    )
                    self._start_job(power_off)
                    av_servers.remove(server)